### DISCUSSION TITLE
Nova::userTimezone throw `Attempt to read property "timezone" on null when user is not logged in` when user is not logged in

### DISCUSSION THREAD

**Discussion by tonyputi** [laravel/nova-issues#5187] [Q&A] [answered]
- Laravel Version: 9.45.1
- Nova Version: 4.19.5
- PHP Version: 8.1.13
- Database Driver & Version: MySQL 8.0 / MariaDB 10.0
- Operating System and Version: Ventura 13.1
- Browser type and version: Chrome 108.0.5359.124
- Reproduction Repository: https://github.com/###/###

### Description:
When user is not yet logged in and `Nova::userTimezone` is used on `NovaServiceProvider` the error `Attempt to read property "timezone" on null` is thrown.

### Detailed steps to reproduce the issue on a fresh Nova installation:

- add a new column on the user table called `timezone`;
- define the following code into the `NovaServiceProvider` boot method:
```
if (! $this->app->runningInConsole()) {
    // Setting user timezone and locale
    Nova::userTimezone(function (Request $request) {
        return $request->user()->timezone;
    });
}
```
- as logged out user try to refresh the page and the error will be thrown

**Comment by crynobone** [comment]
`$request->user()` may return `App\Models\User` or `null`, you just have to consider both scenario.

**Reply by tonyputi** [reply]
ok make sense but then why we have a function called `userTimezone` if user can be null? should then make sense call it `novaTimezone`? What I mean is the word `userTimezone` looks like referring to a logged in user otherwise for which user should be set the timezone? also in the documentation you are suggesting to do something like this... https://nova.laravel.com/docs/4.0/resources/date-fields.html#customizing-the-timezone

**Comment by crynobone** [comment, marked as answer]
Documentation has been updated with the correct example: https://github.com/laravel/nova-docs/pull/499
